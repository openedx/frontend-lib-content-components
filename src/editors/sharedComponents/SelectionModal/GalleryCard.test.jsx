import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { Image } from '@edx/paragon';
import { GalleryCard } from './GalleryCard';

jest.mock('@edx/paragon', () => ({
  ...jest.requireActual('@edx/paragon'),
  Badge: 'Badge',
  SelectableBox: 'SelectableBox',
  Image: 'Image',
}));

describe('GalleryCard component', () => {
  const asset = {
    externalUrl: 'props.img.externalUrl',
    displayName: 'props.img.displayName',
    dateAdded: 12345,
  };
  const thumbnailFallback = (<span>Image failed to load</span>);
  let el;
  beforeEach(() => {
    el = shallow(<GalleryCard asset={asset} />);
  });
  test(`snapshot: dateAdded=${asset.dateAdded}`, () => {
    expect(el.snapshot).toMatchSnapshot();
  });
  it('loads Image with src from image external url', () => {
    expect(el.instance.findByType(Image)[0].props.src).toEqual(asset.externalUrl);
  });
  it('snapshot with thumbnail fallback and load error', () => {
    el = shallow(<GalleryCard asset={asset} thumbnailFallback={thumbnailFallback} />);
    el.instance.findByType(Image)[0].props.onError();
    expect(el.snapshot).toMatchSnapshot();
  });
  it('snapshot with thumbnail fallback and no error', () => {
    el = shallow(<GalleryCard asset={asset} thumbnailFallback={thumbnailFallback} />);
    expect(el.snapshot).toMatchSnapshot();
  });
  it('snapshot with status badge', () => {
    el = shallow(<GalleryCard asset={{ ...asset, status: 'failed', statusBadgeVariant: 'danger' }} />);
    expect(el.snapshot).toMatchSnapshot();
  });
  it('snapshot with duration badge', () => {
    el = shallow(<GalleryCard asset={{ ...asset, duration: 60 }} />);
    expect(el.snapshot).toMatchSnapshot();
  });
  it('snapshot with duration transcripts', () => {
    el = shallow(<GalleryCard asset={{ ...asset, transcripts: ['es'] }} />);
    expect(el.snapshot).toMatchSnapshot();
  });
});
