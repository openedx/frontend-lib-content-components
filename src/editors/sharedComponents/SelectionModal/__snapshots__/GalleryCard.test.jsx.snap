// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCard component snapshot with duration badge 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
      <Badge
        style={
          Object {
            "backgroundColor": "black",
            "bottom": "6px",
            "position": "absolute",
            "right": "6px",
          }
        }
        variant="dark"
      >
        01:00
      </Badge>
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;

exports[`GalleryCard component snapshot with duration transcripts 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <div
        style={
          Object {
            "margin": "0 0 5px 0",
          }
        }
      >
        <injectIntl(ShimmedIntlComponent)
          transcripts={
            Array [
              "es",
            ]
          }
        />
      </div>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;

exports[`GalleryCard component snapshot with status badge 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;

exports[`GalleryCard component snapshot with thumbnail fallback and load error 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        onError={[Function]}
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;

exports[`GalleryCard component snapshot with thumbnail fallback and no error 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        onError={[Function]}
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;

exports[`GalleryCard component snapshot: dateAdded=12345 1`] = `
<ForwardRef
  checked={false}
  className="card bg-white shadow-none border-0 py-0"
  inputHidden={true}
  isIndeterminate={false}
  isInvalid={false}
  key="props.img.externalUrl"
  onClick={[Function]}
  onFocus={[Function]}
  type="radio"
>
  <div
    className="card-div d-flex flex-row flex-nowrap align-items-center"
  >
    <div
      className="position-relative"
      style={
        Object {
          "height": "100px",
          "width": "200px",
        }
      }
    >
      <Image
        src="props.img.externalUrl"
        style={
          Object {
            "border": "none",
            "height": "100px",
            "width": "200px",
          }
        }
      />
    </div>
    <div
      className="card-text px-3 py-2"
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <h3
        className="text-primary-500"
      >
        props.img.displayName
      </h3>
      <p
        className="text-gray-500"
        style={
          Object {
            "fontSize": "11px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Added {date} at {time}"
          description="File date-added string"
          id="authoring.selectionmodal.addedDate.label"
          values={
            Object {
              "date": <FormattedDate
                value={12345}
              />,
              "time": <FormattedTime
                value={12345}
              />,
            }
          }
        />
      </p>
    </div>
  </div>
</ForwardRef>
`;
